---
title: "MDD MWAS - comparison between smoking variables"
author: "Miruna Barbu"
date: "13/05/2021"
output: 
  rmarkdown::html_vignette:
vignette: >
  %\VignetteIndexEntry{MDD MWAS - smoking comparison in GS}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}

---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
htmltools::includeHTML("mdd_mwas_smoke_phenotype_gs.html")
```

```{r load R packages, echo=FALSE, include=FALSE}
# Load packages
library(dplyr)      # Data management
library(pbapply)    # Data management
library(tools)      # Data management
library("readr")    # Data management
library("tibble")   # Data management
library(tidyverse)  # Data management
library(readxl)
library(pander)
# BiocManager::install("ramwas")
library(ramwas)

```

```{r EpiSmoker installation, echo=FALSE, include=FALSE}
# Calculate EpiSmoker

# EpiSmoker dependencies
# library(IlluminaHumanMethylation450kmanifest)
# library(minfi)
# library(htmlTable)
# library(rmarkdown)
# 
# # EpiSmoker installation
# source("http://bioconductor.org/biocLite.R")
# install.packages("devtools") # if you don't have the package, run install.packages("devtools")
# library(devtools)
# install_github("sailalithabollepalli/EpiSmokEr")
library(EpiSmokEr)

```

#### **3 ways to predict smoking using EpiSmoker**

Source article: https://doi.org/10.2217/epi-2019-0206 

There are 3 options to estimate smoking status using EpiSmoker, all derived from whole blood 450K array:

1. **Smoking status**

Uses 121 CpGs to predict smoking status of each individual (LASSO approach). Sex and intercept derived from study are also used in the calculation.

Bollepalli et al. developed EpiSmokEr (Epigenetic Smoking status Estimator), an easy-to-use R package to facilitate the prediction of smoking status. Given methylation data from microarray assays, it provides predicted smoking status as an output. EpiSmokEr can be run not only on whole datasets, but also on isolated individual samples. To be comprehensive, Bollepalli et al. also provide functions for computing SSc and MS.


```{r Smoking status prediction, echo=FALSE, include=FALSE}
# load(system.file("extdata", "SSt_coefficients.rda", package= "EpiSmokEr"))
# knitr::kable(head(SSt_coefficients, caption = "121 CpGs and coefficients selected by Multinomial LASSO approach"))

# Extract the 121 CpGs in a list to then extract from raw methylation data
# smoker_status_cpgs=rownames(SSt_coefficients)
# smoker_status_cpgs=as.data.frame(smoker_status_cpgs)
# smoker_status_cpgs=smoker_status_cpgs[-1,]
# smoker_status_cpgs=as.data.frame(smoker_status_cpgs)
# 
# # Save
# write.table(smoker_status_cpgs, "/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/smoking_status_cpgs.txt", quote=F,row.names=F)

```

N=86/121 of CpGs are available in GS.

2. **Smoking score**

Uses 187 smoking-associated CpGs identified in Zeilinger et al., 2013 (https://doi.org/10.1371/journal.pone.0063812) and calculated using method from Elliott et al., 2014 (https://doi.org/10.1186/1868-7083-6-4) 


Elliot et al. have suggested calculating a weighted score referred to as ‘smoking score’ by adding up the methylation levels of the 187 CpG sites found to be significantly associated with smoking by Zellinger et al., after first multiplying each methylation value by its effect size in Zellinger et al.’s EWAS study.

**Study population in Zeilinger et al.: KORA S4** 

```{r Zeilinger et al. CpGs, echo=FALSE}
# Extract the 187 CpGs in a list to then extract from raw methylation data
# smoke_score_cpgs=EpiSmokEr::Illig_data$cpgs
# smoke_score_cpgs=as.data.frame(smoke_score_cpgs)
# 
# # Save
# write.table(smoke_score_cpgs, "/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/smoking_score_cpgs.txt", quote=F,row.names=F)

```

N=165/187 of CpGs are available in GS.

3. **Methylation score**

Uses 4 smoking-associated CpGs identified in Zhang et al., 2016 (https://doi.org/10.1016/j.envres.2016.01.026) 

Zhang et al. used a regression approach. They first determined significantly associated CpG sites in their cohort, and then employed stepwise logistic regression with forward selection. They used a significance-based stopping criterion and found that already after including the **four most significant loci**, the addition of the fifth locus no longer resulted in a significant reduction of the fit deviance. Therefore, their proposed ‘methylation score’ only requires **four CpG sites**.


```{r Zhang et al. CpGs, echo=FALSE}
# Extract the 4 CpGs in a list to then extract from raw methylation data
# meth_score_cpgs=EpiSmokEr::Zhangetal_cpgs
# meth_score_cpgs=as.data.frame(meth_score_cpgs)
# meth_score_cpgs=rownames(meth_score_cpgs)
# meth_score_cpgs=as.data.frame(meth_score_cpgs)
# 
# # Save
# write.table(meth_score_cpgs, "/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/meth_score_cpgs.txt", quote=F,row.names=F)

```

N=2/4 of CpGs are available in GS.


```{r Calculate scores using EpiSmoker, echo=FALSE, include=FALSE}
# Calculate smoking scores using the comprehensive (all) method from EpiSmoker
# Read in sex
sex=read.table("C:/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/sex.txt",header=T)

# Read in the three beta-value DNAm files for all three scores
smoke_status_beta=readRDS("C:/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/beta_smoke_status.rds")
smoke_status_beta=select(smoke_status_beta,-"203067920168_R08C01")

smoke_score_beta=readRDS("C:/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/beta_smoke_score.rds")
smoke_score_beta=select(smoke_score_beta,-"203067920168_R08C01")

meth_score_beta=readRDS("C:/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/beta_meth_score.rds")
meth_score_beta=select(meth_score_beta,-"203067920168_R08C01")

# Run EpiSmoker
result_All <- epismoker(dataset_QN=smoke_status_beta, dataset_ILM=meth_score_beta, dataset_SQN=smoke_score_beta, samplesheet = sex, method = "all")

```

#### Descriptive statistics

```{r Descr stats, echo=FALSE}
# TABLES

# PREP DATA
# load("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Other work activities/Cohort data dictionaries/masterDB.Rdata")
# ever_smoke=data.frame(totaldata[,c(1,44)])
# linker=readRDS("C:/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/Complex trait MRS/Analysis/Scripts & datasets for MRS calculation/participant_ids.rds")
# colnames(linker)[2]<-"id"
# ever_smoke=merge(ever_smoke,linker,by="id")
# colnames(ever_smoke)[3]<-"SampleName"
# results_smoke=merge(result_all,ever_smoke,by="SampleName")
# results_smoke$smoker_data[results_smoke$ever_smoke==1]<-"Current smoker"
# results_smoke$smoker_data[results_smoke$ever_smoke==2]<-"Former smoker (stopped < 12 months)"
# results_smoke$smoker_data[results_smoke$ever_smoke==3]<-"Former smoker (stopped > 12 months)"
# results_smoke$smoker_data[results_smoke$ever_smoke==4]<-"Never smoker"
# 
# results_smoke=results_smoke[complete.cases(results_smoke),]
# # N=9,825 with smoking data
# # Save
# saveRDS(results_smoke, "/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/results_smoke.rds")

library(dplyr)
results_smoke=readRDS("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/results_smoke.rds")

# CREATE TABLES USING DPLYR AND THEN SAVE THESE
# Smoke status
# table_smoke_status=results_smoke %>% count(`GS smoking status`=smoker_data,`Predicted smoking status`=PredictedSmokingStatus)

# saveRDS(table_smoke_status,"/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/table_smoke_status.rds")

table_smoke_status=readRDS("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/table_smoke_status.rds")

# Smoking score
# table_smoke_score=results_smoke %>% group_by(`Smoking status`=smoker_data) %>% summarize(`Smoking score mean` = mean(smokingScore))
# 
# saveRDS(table_smoke_score,"/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/table_smoke_score.rds")

table_smoke_score=readRDS("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/table_smoke_score.rds")

# Methylation score
# table_meth_score=results_smoke %>% group_by(`Smoking status`=smoker_data) %>% summarize(`Methylation score mean` = mean(methylationScore))
# 
# saveRDS(table_meth_score,"/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/table_meth_score.rds")

table_meth_score=readRDS("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/table_meth_score.rds")

# Put tables together into one
table_final=cbind(table_smoke_status,table_smoke_score,table_meth_score)
table_final=data.frame(table_final[,c(1,3,2,5,7)])

# Print table
knitr::kable(table_final, caption="Predicted smoking status, mean smoking score, and mean methylation score for current, former, and never smokers in GS.")

```

#### Possible issues with calculating these in different cohorts

- M-values required for MWAS, but beta values required for EpiSmoker calculation
- Different cohorts will have different CpGs available for calculation
- Required to extract each set of CpGs for Epismoker

```{r Rscript prep, echo=FALSE, include=FALSE}
# Rscript to extract CpGs from DNAm 450K data in GS
# Load DNAm data
# Read in all 3 CpG lists
# extract using subset, e.g.
# mval_smokescore=subset(mvals, rownames(mvals) %in% smokescore$cpgs)
# Transform subset of mvals to beta
# library(lumi)
# m2beta(mval_smokescore)

# Extract sex from GS and input DNAm IDs as rownames in this file, for calculation of smoke status predictor
# load("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Other work activities/Cohort data dictionaries/masterDB.Rdata")
# sex=data.frame(totaldata[,c(1,82)])
# sex$sex=ifelse(sex$sex=="M", 1, 2)
# colnames(sex)[1]<-"gwas"
# # Read in DNAm linker
# linker=readRDS("C:/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/Complex trait MRS/Analysis/Scripts & datasets for MRS calculation/participant_ids.rds")
# 
# # Merge with sex
# sex=merge(sex, linker,by="gwas")
# rownames(sex)<-sex$ID
# sex2=data.frame(sex[,2])
# rownames(sex2)<-rownames(sex)
# colnames(sex2)[1]<-"sex"
# 
# # Check which individual is not present in sex from linker
# notinsex=subset(linker, !(ID %in% sex$ID))
# # DNAm: 203067920168_R08C01 ; GS ID: 18984
# 
# # Save
# write.table(sex2, "C:/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/sex.txt", quote=F,row.names=T)

```

#### **Comparing smoking signatures across MWAS - MDD**

5 basic models were run, with technical and biological covariates accounted for. CpGs were included as outcomes and MDD diagnosis as the predictor variable, with:

(1) no smoking covariate

(2) phenotypic smoking (current, former, never smokers)

(3) AHRR probe (cg05575921)

(4) EpiSmoker smoking-derived score (based on 165/187 CpGs)

(5) EpiSmoker methylation-derived smoking score (based on 2/4 CpGs). 

```{r Comparing different MWAS, echo=FALSE}
# Read in basic MWAS - N pvals=57
basic_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/mdd_mwas_may2021/mdd_basic_model_smokers.toptable.txt")

# P-value correction
basic_mwas$p.corrected=p.adjust(basic_mwas$P.Value,method="bonferroni")
basic_mwas_corr=subset(basic_mwas, P.Value < 0.000000036)

# Order by most significant
basic_mwas_corr=basic_mwas_corr[order(basic_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
basic_mwas_corr=data.frame(basic_mwas_corr[,c(1,2,3,14,10,17)])
basic_mwas_corr$geneSymbol = sub("\\;.*", "", basic_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(basic_mwas_corr,row.names = F)


# Read in smoking phenotype MWAS - N pvals=15
smokingpheno_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/Results/basic_model_phenotype_smoke.toptable.txt")

# P-value correction
smokingpheno_mwas$p.corrected=p.adjust(smokingpheno_mwas$P.Value,method="bonferroni")
smokingpheno_mwas_corr=subset(smokingpheno_mwas, P.Value < 0.000000036)

# Order by most significant
smokingpheno_mwas_corr=smokingpheno_mwas_corr[order(smokingpheno_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
smokingpheno_mwas_corr=data.frame(smokingpheno_mwas_corr[,c(1,2,3,14,10,17)])
smokingpheno_mwas_corr$geneSymbol = sub("\\;.*", "", smokingpheno_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smokingpheno_mwas_corr,row.names = F)


# Read in smoking AHRR probe MWAS - N pvals=26
smokingahrr_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/Results/basic_model_ahrr_smoke.toptable.txt")

# P-value correction
smokingahrr_mwas$p.corrected=p.adjust(smokingahrr_mwas$P.Value,method="bonferroni")
smokingahrr_mwas_corr=subset(smokingahrr_mwas, P.Value < 0.000000036)

# Order by most significant
smokingahrr_mwas_corr=smokingahrr_mwas_corr[order(smokingahrr_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
smokingahrr_mwas_corr=data.frame(smokingahrr_mwas_corr[,c(1,2,3,14,10,17)])
smokingahrr_mwas_corr$geneSymbol = sub("\\;.*", "", smokingahrr_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smokingahrr_mwas_corr,row.names = F)

# Read in smoking phenotype MWAS - N pvals=42
smoking_smokescore_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/Results/basic_model_smokescore_smoke.toptable.txt")

# P-value correction
smoking_smokescore_mwas$p.corrected=p.adjust(smoking_smokescore_mwas$P.Value,method="bonferroni")
smoking_smokescore_mwas_corr=subset(smoking_smokescore_mwas, P.Value < 0.000000036)

# Order by most significant
smoking_smokescore_mwas_corr=smoking_smokescore_mwas_corr[order(smoking_smokescore_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
smoking_smokescore_mwas_corr=data.frame(smoking_smokescore_mwas_corr[,c(1,2,3,14,10,17)])
smoking_smokescore_mwas_corr$geneSymbol = sub("\\;.*", "", smoking_smokescore_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smoking_smokescore_mwas_corr,row.names = F)

# Read in smoking phenotype MWAS - N pvals=46
smoking_methscore_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/Results/basic_model_methscore_smoke.toptable.txt")

# P-value correction
smoking_methscore_mwas$p.corrected=p.adjust(smoking_methscore_mwas$P.Value,method="bonferroni")
smoking_methscore_mwas_corr=subset(smoking_methscore_mwas, P.Value < 0.000000036)

# Order by most significant
smoking_methscore_mwas_corr=smoking_methscore_mwas_corr[order(smoking_methscore_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
smoking_methscore_mwas_corr=data.frame(smoking_methscore_mwas_corr[,c(1,2,3,14,10,17)])
smoking_methscore_mwas_corr$geneSymbol = sub("\\;.*", "", smoking_methscore_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smoking_methscore_mwas_corr,row.names = F)

```

#### **Comparing statistics across MWAS - MDD**

```{r QQ plots for each model, echo=FALSE}
qqPlotFast(basic_mwas$P.Value)
title('QQ-plot\nNo smoking covariate')
qqPlotFast(smokingpheno_mwas$P.Value)
title('QQ-plot\nPhenotypic smoking covariate')
qqPlotFast(smokingahrr_mwas$P.Value)
title('QQ-plot\nAHRR probe (cg05575921) covariate')
qqPlotFast(smoking_smokescore_mwas$P.Value)
title('QQ-plot\nEpiSmoker: smoke score covariate')
qqPlotFast(smoking_methscore_mwas$P.Value)
title('QQ-plot\nEpiSmoker: methylation score covariate')

# Create table with each method & lambda
lambda=matrix(c("No smoking covariate","Phenotypic smoking","AHRR probe (cg05575921)","EpiSmoker smoke score", "EpiSmoker methylation score",1.490,1.374,1.396,1.417,1.492,57,15,26,42,46,min(basic_mwas_corr$P.Value),min(smokingpheno_mwas_corr$P.Value),min(smokingahrr_mwas_corr$P.Value),min(smoking_smokescore_mwas_corr$P.Value),min(smoking_methscore_mwas_corr$P.Value),min(basic_mwas_corr$beta),min(smokingpheno_mwas_corr$beta),min(smokingahrr_mwas_corr$beta),min(smoking_smokescore_mwas_corr$beta),min(smoking_methscore_mwas_corr$beta),max(basic_mwas_corr$beta),max(smokingpheno_mwas_corr$beta),max(smokingahrr_mwas_corr$beta),max(smoking_smokescore_mwas_corr$beta),max(smoking_methscore_mwas_corr$beta)), nrow=5,ncol=6)
colnames(lambda)[1:6]<-c("Covariate included in MWAS","lambda","N Bonferroni-corrected CpGs","Min p-value","Min effect size", "Max effect size")
lambda=as.data.frame(lambda)
lambda=lambda[order(lambda$lambda),]

lambda$`Min effect size`=as.numeric(lambda$`Min effect size`)
lambda$`Min effect size`=round(lambda$`Min effect size`,digits=3)
lambda$`Max effect size`=as.numeric(lambda$`Max effect size`)
lambda$`Max effect size`=round(lambda$`Max effect size`,digits=3)

knitr::kable(lambda,caption="Variance inflation factor, number of Bonferroni-corrected CpGs, minimum p-value, and minimum & maximum effect sizes for every MWAS based on smoking covariate included. All models are corrected for technical and biological covariates.",row.names = F)

```

#### **Correlation between AHRR probe (cg05575921) and EpiSmoker smoking score**

```{r Correlation between AHRR and epismoker score, echo=FALSE}
smoke_score=data.frame(result_All[,c(1,2)])
ahrr=t(meth_score_beta)
ahrr=as.data.frame(ahrr)
ahrr$SampleName=rownames(ahrr)
ahrr_smokescore=merge(smoke_score,ahrr,by="SampleName")
ahrr_smokescore=ahrr_smokescore[complete.cases(ahrr_smokescore),]

cor(ahrr_smokescore$smokingScore,ahrr_smokescore$cg05575921)
plot(ahrr_smokescore$smokingScore,ahrr_smokescore$cg05575921)

```

#### Check MDD MWAS results against Joehanes et al. FDR-corrected CpGs 

```{r Smoking CpGs in Joehanes et al., echo=FALSE}
# Read in Joehanes et al. FDR-corrected CpGs, N=18,760
joehanes=read.csv("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/Complex trait MRS/Analysis/ewas_weights/smoking_joehanescsv.csv")

# Read in all 5 MWAS MDD models
# Read in basic MWAS - N pvals=57
basic_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/mdd_mwas_may2021/mdd_basic_model_smokers.toptable.txt")

# P-value correction
basic_mwas$p.corrected=p.adjust(basic_mwas$P.Value,method="bonferroni")
basic_mwas_corr=subset(basic_mwas, P.Value < 0.000000036)

# Order by most significant
basic_mwas_corr=basic_mwas_corr[order(basic_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
basic_mwas_corr=data.frame(basic_mwas_corr[,c(1,2,3,14,10,17)])
basic_mwas_corr$geneSymbol = sub("\\;.*", "", basic_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(basic_mwas_corr,row.names = F)


# Read in smoking phenotype MWAS - N pvals=15
smokingpheno_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/Results/basic_model_phenotype_smoke.toptable.txt")

# P-value correction
smokingpheno_mwas$p.corrected=p.adjust(smokingpheno_mwas$P.Value,method="bonferroni")
smokingpheno_mwas_corr=subset(smokingpheno_mwas, P.Value < 0.000000036)

# Order by most significant
smokingpheno_mwas_corr=smokingpheno_mwas_corr[order(smokingpheno_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
smokingpheno_mwas_corr=data.frame(smokingpheno_mwas_corr[,c(1,2,3,14,10,17)])
smokingpheno_mwas_corr$geneSymbol = sub("\\;.*", "", smokingpheno_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smokingpheno_mwas_corr,row.names = F)


# Read in smoking AHRR probe MWAS - N pvals=26
smokingahrr_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/Results/basic_model_ahrr_smoke.toptable.txt")

# P-value correction
smokingahrr_mwas$p.corrected=p.adjust(smokingahrr_mwas$P.Value,method="bonferroni")
smokingahrr_mwas_corr=subset(smokingahrr_mwas, P.Value < 0.000000036)

# Order by most significant
smokingahrr_mwas_corr=smokingahrr_mwas_corr[order(smokingahrr_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
smokingahrr_mwas_corr=data.frame(smokingahrr_mwas_corr[,c(1,2,3,14,10,17)])
smokingahrr_mwas_corr$geneSymbol = sub("\\;.*", "", smokingahrr_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smokingahrr_mwas_corr,row.names = F)

# Read in smoking phenotype MWAS - N pvals=42
smoking_smokescore_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/Results/basic_model_smokescore_smoke.toptable.txt")

# P-value correction
smoking_smokescore_mwas$p.corrected=p.adjust(smoking_smokescore_mwas$P.Value,method="bonferroni")
smoking_smokescore_mwas_corr=subset(smoking_smokescore_mwas, P.Value < 0.000000036)

# Order by most significant
smoking_smokescore_mwas_corr=smoking_smokescore_mwas_corr[order(smoking_smokescore_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
smoking_smokescore_mwas_corr=data.frame(smoking_smokescore_mwas_corr[,c(1,2,3,14,10,17)])
smoking_smokescore_mwas_corr$geneSymbol = sub("\\;.*", "", smoking_smokescore_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smoking_smokescore_mwas_corr,row.names = F)

# Read in smoking phenotype MWAS - N pvals=46
smoking_methscore_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/Results/basic_model_methscore_smoke.toptable.txt")

# P-value correction
smoking_methscore_mwas$p.corrected=p.adjust(smoking_methscore_mwas$P.Value,method="bonferroni")
smoking_methscore_mwas_corr=subset(smoking_methscore_mwas, P.Value < 0.000000036)

# Order by most significant
smoking_methscore_mwas_corr=smoking_methscore_mwas_corr[order(smoking_methscore_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
smoking_methscore_mwas_corr=data.frame(smoking_methscore_mwas_corr[,c(1,2,3,14,10,17)])
smoking_methscore_mwas_corr$geneSymbol = sub("\\;.*", "", smoking_methscore_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smoking_methscore_mwas_corr,row.names = F)

# Check overlap between CpGs
basic_joeh=subset(basic_mwas_corr, ID %in% joehanes$CpG) # N=15
pheno_joeh=subset(smokingpheno_mwas_corr, ID %in% joehanes$CpG) # N=0
ahrr_joeh=subset(smokingahrr_mwas_corr, ID %in% joehanes$CpG) # N=0
smokescore_joeh=subset(smoking_smokescore_mwas_corr, ID %in% joehanes$CpG) # N=0
methscore_joeh=subset(smoking_methscore_mwas_corr, ID %in% joehanes$CpG) # N=9

# Create table with each method & lambda
overlap_cpgs=matrix(c("No smoking covariate","Phenotypic smoking","AHRR probe (cg05575921)","EpiSmoker smoke score", "EpiSmoker methylation score",57,15,26,42,46,min(basic_mwas_corr$P.Value),min(smokingpheno_mwas_corr$P.Value),min(smokingahrr_mwas_corr$P.Value),min(smoking_smokescore_mwas_corr$P.Value),min(smoking_methscore_mwas_corr$P.Value),15,0,0,0,9), nrow=5,ncol=4)
colnames(overlap_cpgs)[1:4]<-c("Covariate included in MWAS","N Bonferroni-corrected CpGs","Min p-value","Overlapping CpGs with Joehanes et al.")
overlap_cpgs=as.data.frame(overlap_cpgs)
overlap_cpgs=overlap_cpgs[order(overlap_cpgs$`N Bonferroni-corrected CpGs`),]

knitr::kable(overlap_cpgs,caption="Number of Bonferroni-corrected CpGs, minimum p-value, and overlapping top CpGs with Joehanes et al. FDR-corrected CpGs for every MDD MWAS based on smoking covariate included. All models are corrected for technical and biological covariates.",row.names = F)

```

#### MWAS for trait associated with smoking - educational attainment

```{r Educ Attain MWAS for smoking check,echo=FALSE,include=FALSE}
# Read in masterDB to check educational attainment
# load("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Other work activities/Cohort data dictionaries/masterDB.Rdata")
# educ=totaldata[,c(1,42,3)]
# # Turn qualification into ordinal variable
# # 1 - College or University degree
# # 2 - Other professional or technical qualification
# # 3 - NVQ or HND or HNC or equivalent
# # 4 - Higher Grade, A levels, AS levels or equivalent
# # 5 - Standard Grade, O levels, GCSEs or equivalent
# # 6 - CSEs or equivalent
# # 7 - School leavers certificate
# # 8 - Other
# # 9 - No Qualification
# 
# educ$ordered_qualification <- factor(educ$qualification, order = TRUE, levels = c(8,9,7,6,5,4,3,2,1))
# 
# # Print the result to the console
# educ$ordered_qualification
# Levels: 8 < 9 < 7 < 6 < 5 < 4 < 3 < 2 < 1
# 
# # Save this phenotype to run MWAS with
# pdata_educ=educ[,c(1,4)]
# write.csv(pdata_educ, "/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/educ_comparison/pdata_educ.csv",quote=F,row.names=F)

# Need to run 5 models: basic (no covariates), phenotypic smoke, ahrr probe, epismoker smoke score, epismoker meth score

```

#### **Comparing statistics across MWAS**

```{r Educational attainment stats comparison, echo=FALSE}
# Read in basic MWAS - N pvals=6097
educ_basic_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/educ_comparison/educ_basic_model_nocovs.toptable.txt")

# P-value correction
educ_basic_mwas$p.corrected=p.adjust(educ_basic_mwas$P.Value,method="bonferroni")
educ_basic_mwas_corr=subset(educ_basic_mwas, P.Value < 0.000000036)

# Order by most significant
educ_basic_mwas_corr=educ_basic_mwas_corr[order(educ_basic_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
educ_basic_mwas_corr=data.frame(educ_basic_mwas_corr[,c(1,2,3,14,10,17)])
educ_basic_mwas_corr$geneSymbol = sub("\\;.*", "", educ_basic_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(basic_mwas_corr,row.names = F)

# Read in smoking phenotype MWAS - N pvals=1468
educ_smokingpheno_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/educ_comparison/educ_basic_model_phenosmoke.toptable.txt")

# P-value correction
educ_smokingpheno_mwas$p.corrected=p.adjust(educ_smokingpheno_mwas$P.Value,method="bonferroni")
educ_smokingpheno_mwas_corr=subset(educ_smokingpheno_mwas, P.Value < 0.000000036)

# Order by most significant
educ_smokingpheno_mwas_corr=educ_smokingpheno_mwas_corr[order(educ_smokingpheno_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
educ_smokingpheno_mwas_corr=data.frame(educ_smokingpheno_mwas_corr[,c(1,2,3,14,10,17)])
educ_smokingpheno_mwas_corr$geneSymbol = sub("\\;.*", "", educ_smokingpheno_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smokingpheno_mwas_corr,row.names = F)


# Read in smoking AHRR probe MWAS - N pvals=2129
educ_smokingahrr_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/educ_comparison/educ_basic_model_ahrrprobe.toptable.txt")

# P-value correction
educ_smokingahrr_mwas$p.corrected=p.adjust(educ_smokingahrr_mwas$P.Value,method="bonferroni")
educ_smokingahrr_mwas_corr=subset(educ_smokingahrr_mwas, P.Value < 0.000000036)

# Order by most significant
educ_smokingahrr_mwas_corr=educ_smokingahrr_mwas_corr[order(educ_smokingahrr_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
educ_smokingahrr_mwas_corr=data.frame(educ_smokingahrr_mwas_corr[,c(1,2,3,14,10,17)])
educ_smokingahrr_mwas_corr$geneSymbol = sub("\\;.*", "", educ_smokingahrr_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smokingahrr_mwas_corr,row.names = F)

# Read in smoking phenotype MWAS - N pvals=4182
educ_smoking_smokescore_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/educ_comparison/educ_basic_model_epi_smokescore.toptable.txt")

# P-value correction
educ_smoking_smokescore_mwas$p.corrected=p.adjust(educ_smoking_smokescore_mwas$P.Value,method="bonferroni")
educ_smoking_smokescore_mwas_corr=subset(educ_smoking_smokescore_mwas, P.Value < 0.000000036)

# Order by most significant
educ_smoking_smokescore_mwas_corr=educ_smoking_smokescore_mwas_corr[order(educ_smoking_smokescore_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
educ_smoking_smokescore_mwas_corr=data.frame(educ_smoking_smokescore_mwas_corr[,c(1,2,3,14,10,17)])
educ_smoking_smokescore_mwas_corr$geneSymbol = sub("\\;.*", "", educ_smoking_smokescore_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smoking_smokescore_mwas_corr,row.names = F)

# Read in smoking phenotype MWAS - N pvals=4042
educ_smoking_methscore_mwas=read.delim("/Users/mirun/OneDrive - University of Edinburgh/Work directories/Current projects/MDD MWAS meta-analysis/MDD MWAS smoking comparison/educ_comparison/educ_basic_model_epi_methscore.toptable.txt")

# P-value correction
educ_smoking_methscore_mwas$p.corrected=p.adjust(educ_smoking_methscore_mwas$P.Value,method="bonferroni")
educ_smoking_methscore_mwas_corr=subset(educ_smoking_methscore_mwas, P.Value < 0.000000036)

# Order by most significant
educ_smoking_methscore_mwas_corr=educ_smoking_methscore_mwas_corr[order(educ_smoking_methscore_mwas_corr$p.corrected),]

# Present CpG, gene, beta-value, p-value, and p-corrected value
educ_smoking_methscore_mwas_corr=data.frame(educ_smoking_methscore_mwas_corr[,c(1,2,3,14,10,17)])
educ_smoking_methscore_mwas_corr$geneSymbol = sub("\\;.*", "", educ_smoking_methscore_mwas_corr$geneSymbol) # print only first gene symbol (they are repeated in most cases, e.g. "SLC6A4; SLC6A4)

# Print
# pander(smoking_methscore_mwas_corr,row.names = F)


```


```{r QQ plots for each model in educ model, echo=FALSE}
qqPlotFast(educ_basic_mwas$P.Value)
title('QQ-plot\nNo smoking covariate - educational attainment')
qqPlotFast(educ_smokingpheno_mwas$P.Value)
title('QQ-plot\nPhenotypic smoking covariate - educational attainment')
qqPlotFast(educ_smokingahrr_mwas$P.Value)
title('QQ-plot\nAHRR probe (cg05575921) covariate - educational attainment')
qqPlotFast(educ_smoking_smokescore_mwas$P.Value)
title('QQ-plot\nEpiSmoker: smoke score covariate - educational attainment')
qqPlotFast(educ_smoking_methscore_mwas$P.Value)
title('QQ-plot\nEpiSmoker: methylation score covariate - educational attainment')

# Create table with each method & lambda
lambda=matrix(c("No smoking covariate","Phenotypic smoking","AHRR probe (cg05575921)","EpiSmoker smoke score", "EpiSmoker methylation score",2.195,1.755,1.967,2.464,2.216,6097,1468,2129,4182,4042,min(educ_basic_mwas_corr$P.Value),min(educ_smokingpheno_mwas_corr$P.Value),min(educ_smokingahrr_mwas_corr$P.Value),min(educ_smoking_smokescore_mwas_corr$P.Value),min(educ_smoking_methscore_mwas_corr$P.Value),min(educ_basic_mwas_corr$beta),min(educ_smokingpheno_mwas_corr$beta),min(educ_smokingahrr_mwas_corr$beta),min(educ_smoking_smokescore_mwas_corr$beta),min(educ_smoking_methscore_mwas_corr$beta),max(educ_basic_mwas_corr$beta),max(educ_smokingpheno_mwas_corr$beta),max(educ_smokingahrr_mwas_corr$beta),max(educ_smoking_smokescore_mwas_corr$beta),max(educ_smoking_methscore_mwas_corr$beta)), nrow=5,ncol=6)
colnames(lambda)[1:6]<-c("Covariate included in MWAS","lambda","N Bonferroni-corrected CpGs","Min p-value","Min effect size", "Max effect size")
lambda=as.data.frame(lambda)
lambda=lambda[order(lambda$lambda),]

lambda$`Min effect size`=as.numeric(lambda$`Min effect size`)
lambda$`Min effect size`=round(lambda$`Min effect size`,digits=3)
lambda$`Max effect size`=as.numeric(lambda$`Max effect size`)
lambda$`Max effect size`=round(lambda$`Max effect size`,digits=3)

knitr::kable(lambda,caption="Variance inflation factor, number of Bonferroni-corrected CpGs, minimum p-value, and minimum & maximum effect sizes for every Educational Attainment MWAS based on smoking covariate included. All models are corrected for technical and biological covariates.",row.names = F)

```




